python/src/CMakeLists.txt

```diff
@@ -29,8 +29,21 @@ if(NOT MLX_PYTHON_BINDINGS_OUTPUT_DIRECTORY)
   set(MLX_PYTHON_BINDINGS_OUTPUT_DIRECTORY ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
 endif()
 
-set_target_properties(core PROPERTIES LIBRARY_OUTPUT_DIRECTORY
-                                      ${MLX_PYTHON_BINDINGS_OUTPUT_DIRECTORY})
+if(MSVC)
+  set_target_properties(
+    core
+    PROPERTIES LIBRARY_OUTPUT_DIRECTORY_RELEASE
+               ${MLX_PYTHON_BINDINGS_OUTPUT_DIRECTORY}
+               LIBRARY_OUTPUT_DIRECTORY_DEBUG
+               ${MLX_PYTHON_BINDINGS_OUTPUT_DIRECTORY}
+               LIBRARY_OUTPUT_DIRECTORY_RELWITHDEBINFO
+               ${MLX_PYTHON_BINDINGS_OUTPUT_DIRECTORY}
+               LIBRARY_OUTPUT_DIRECTORY_MINSIZEREL
+               ${MLX_PYTHON_BINDINGS_OUTPUT_DIRECTORY})
+else()
+  set_target_properties(core PROPERTIES LIBRARY_OUTPUT_DIRECTORY
+                                        ${MLX_PYTHON_BINDINGS_OUTPUT_DIRECTORY})
+endif()
```

Comment:
Why not just use the first condition for every platform and discard the check for MSVC?

Reply:
Updated. It seems that some build systems other than MSVC also require this to work.